order_analysis.rb

```diff
@@ -24,13 +27,56 @@ def extract_customer(line)
       return
     end
 
-    customer_id = parts[0]
-    first_name = parts[1]
-    last_name = parts[2]
-    location = parts[3]
+    customer = Customer.new(parts[0], parts[1], parts[2], parts[3])
+    @customers.push(customer)
 
-    puts "Customer #{customer_id}: #{first_name} #{last_name} (#{location})"
+  end
+
+  def analyze_orders
+    orders_csv = File.read("orders.csv")
+    orders_lines = orders_csv.split("\n")
```

Comment:
You could try to use the CSV standard library (but maybe do it in a new pull request)

Reply:
that's the next thing on my to do list :)